webapp/Connector/src/utility/PlotTooltip.js

```diff
@@ -0,0 +1,471 @@
+    {
+        var content = this.buildBinTooltip(plot, datas, plotData);
+        ChartUtils.showDataTooltipCallout(content, point, 'hidetooltipmsg', plotName===plot.yGutterName, plotName===plot.xGutterName, plot);
+
+    },
+
+    getStudyLabels: function(records)
+    {
+        var studyLabelsObj = {};
+        Ext.each(records, function(record){
+            studyLabelsObj[record.Study] = true;
+        });
+        return Object.keys(studyLabelsObj);
+    },
+
+    getRawDataRows: function(plot, datas)
+    {
+        var rawRecords = [];
+        Ext.each(datas, function(data) {
+            var d = data.data;
+            rawRecords.push(plot.allDataRowsMap[d.rowKey]);
+        }, this);
+        return rawRecords;
+    },
+
+    loadPointTooltipData: function(plot, point, data, plotName)
+    {
+        var rawRow = plot.allDataRowsMap[data.rowKey];
+        var participantSeq = rawRow[QueryUtils.SUBJECT_SEQNUM_ALIAS];
+        var container = rawRow[QueryUtils.CONTAINER_ALIAS];
+        LABKEY.Query.selectRows({
+            schemaName: 'cds',
+            queryName: 'gridbase',
+            filterArray: [
+                LABKEY.Filter.create('ParticipantSequenceNum', participantSeq),
+                LABKEY.Filter.create('Container', container)
+            ],
+            success: function (results) {
+                var plotData = results.rows[0];
+                var studyLabel = plotData.Study;
+                if (plot.studyMap)
+                {
+                    plotData.StudyShortName = plot.studyMap[studyLabel];
+                    this.showPointTooltip(plot, point, data, plotName, plotData);
+                }
+                else
+                {
+                    LABKEY.Query.selectRows({
+                        schemaName: 'cds',
+                        queryName: 'study',
+                        success: function (studyData) {
+                            var studyRows = studyData.rows;
+                            var studyMap = {};
+                            Ext.each(studyRows, function (row) {
+                                studyMap[row.label] = row.short_name;
+                            });
+                            plot.studyMap = studyMap;
+                            plotData.StudyShortName = plot.studyMap[studyLabel];
+                            this.showPointTooltip(plot, point, data, plotName, plotData);
+                        },
+                        scope: this
+                    });
+                }
+            },
+            scope: this
+        });
+    },
+
+    showPointTooltip: function(plot, point, data, plotName, plotData)
+    {
+        var content = this.buildPointTooltip(plot, data, plotData);
+        ChartUtils.showDataTooltipCallout(content, point, 'hidetooltipmsg', plotName===plot.yGutterName, plotName===plot.xGutterName, plot);
+    },
+
+    buildPointTooltip: function(plot, data, plotData) {
```

Comment:
Minor changes made to buildPointTooltip to use queried plotData instead of originally loaded 'data' for generating tooltip.